MeshLib/MeshGenerators/VtkMeshConverter.cpp

```diff
@@ -356,8 +358,70 @@ MeshLib::Mesh* VtkMeshConverter::convertUnstructuredGrid(vtkUnstructuredGrid* gr
 		elements[i] = elem;
 	}
 
-	return new MeshLib::Mesh(mesh_name, nodes, elements);
+	MeshLib::Mesh* mesh = new MeshLib::Mesh(mesh_name, nodes, elements);
+	convertScalarArrays(*grid, *mesh);
+
+	return mesh;
+}
+
+void VtkMeshConverter::convertScalarArrays(vtkUnstructuredGrid &grid, MeshLib::Mesh &mesh)
+{
+	vtkPointData* point_data = grid.GetPointData();
+	if (point_data != nullptr)
+	{
+		unsigned const n_point_arrays = static_cast<unsigned>(point_data->GetNumberOfArrays());
+		for (unsigned i=0; i<n_point_arrays; ++i)
+			convertArray(point_data->GetArray(i), mesh.getProperties(), MeshLib::MeshItemType::Node);
+	}
+
+	vtkCellData* cell_data = grid.GetCellData();
+	if (cell_data == nullptr)
+		return;
+	unsigned const n_cell_arrays = static_cast<unsigned>(cell_data->GetNumberOfArrays());
+	for (unsigned i=0; i<n_cell_arrays; ++i)
+		convertArray(cell_data->GetArray(i), mesh.getProperties(), MeshLib::MeshItemType::Cell);
+}
+
+void VtkMeshConverter::convertArray(vtkDataArray* array, MeshLib::Properties &properties, MeshLib::MeshItemType type)
+{
+	vtkIdType const nTuples (array->GetNumberOfTuples());
+	int const nComponents (array->GetNumberOfComponents());
+	char const*const array_name (array->GetName());
+
+	vtkDoubleArray* double_array = vtkDoubleArray::SafeDownCast(array);
+	if (double_array)
+	{
+		boost::optional<MeshLib::PropertyVector<double> &> vec
+			(properties.createNewPropertyVector<double>(array_name, type, nComponents));
+		if (!vec)
```

Comment:
under which circumstances could conversion of a ```vtkDoubleArray``` to a property vector fail? (i.e. is this test necessary?)